lib/labels.js

```diff
@@ -0,0 +1,258 @@
+  }
+
+  // Get single label or all labels if `labelName` is not specified
+  // from `repo`
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   labelName <string>
+  async getFrom(repo, labelName) {
+    let path = this[getPath](repo, labelName);
+    if (labelName) {
+      const { data } = await this[request](path, 'GET');
+      return JSON.parse(data);
+    }
+    const result = [];
+    path += '?per_page=100';
+    while (path) {
+      const { data, res } = await this[request](path, 'GET');
+      result.push(...JSON.parse(data));
+
+      if (!res.headers.link) break;
+      path = parseLinkHeader(res.headers.link).get('next');
+    }
+    return result;
+  }
+
+  // Get single label or all labels if `labelName` is not specified
+  // from current repository
+  //   labelName <string>
+  async get(labelName) {
+    return this.getFrom(this.options.repo, labelName);
+  }
+
+  // Create label or labels in `repoOwner/repoName`
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   label <Object> | <Object[]>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>, optional
+  async createIn(repo, label) {
+    const path = this[getPath](repo);
+    if (Array.isArray(label)) {
+      return Promise.all(
+        label.map(label => this[request](path, 'POST', JSON.stringify(label)))
+      ).map(({ data }) => JSON.parse(data));
+    } else {
+      const { data } = await this[request](path, 'POST', JSON.stringify(label));
+      return JSON.parse(data);
+    }
+  }
+
+  // Create label or labels in current repository
+  //   label <Object> | <Object[]>
+  //     name <string>
+  //     color <string>, hexadecimal color code, e.g. f29513
+  //     description <string>, optional
+  async create(label) {
+    return this.createIn(this.options.repo, label);
+  }
+
+  // Get used labels from `repo`
+  //   repo <string> e.g. `repoOwner/repoName`
+  async getUsedLabels(repo) {
+    const { data } = await this[request](
+      `/repos/${repo}/issues?state=all`,
+      'GET'
+    );
+    const result = new Map();
+    JSON.parse(data).forEach(issue =>
+      issue.labels.forEach(label => {
+        const name = label.url.split('/').pop();
+        if (!result.has(name)) {
+          result.set(
+            name,
+            `https://github.com/${repo}/issues?q=label%3A${name}`
+          );
+        }
+      })
+    );
+    return result;
+  }
+
+  // Delete label or labels in `repoOwner/repoName`
+  //   repo <string> e.g. `repoOwner/repoName`
+  //   labelName <string> | <string[]>, optional, if not specified, all labels
+  //       will be deleted
+  async deleteIn(repo, labelName) {
+    if (!labelName) {
+      labelName = (await this.getFrom(repo)).map(label => label.name);
+    }
+    if (!Array.isArray(labelName)) labelName = [labelName];
+
+    const usedLabels = await this.getUsedLabels(repo);
+    if (usedLabels.size) {
+      console.error(
```

Comment:
I don't think this should print to stdout/err, we can just return the status of all labels from this method for the caller to decide what to do with the skipped ones.

Perhaps this should be configurable `deleteUsed = false`.